src/mbgl/gfx/drawable.cpp

```diff
@@ -6,13 +6,20 @@
 #include <mbgl/gfx/index_vector.hpp>
 #include <mbgl/gfx/types.hpp>
 #include <mbgl/renderer/render_pass.hpp>
+#include <mbgl/renderer/render_tile.hpp>
 
 namespace mbgl {
 namespace gfx {
 
 struct Drawable::Impl {
     gfx::ColorMode colorMode = gfx::ColorMode::disabled();
     gfx::CullFaceMode cullFaceMode = gfx::CullFaceMode::disabled();
+
+    std::shared_ptr<Bucket> bucket;
+    PaintPropertyBindersBase* binders = nullptr; // owned by `bucket`
```

Comment:
Why not get it from the bucket when needed?

Reply:
Just to avoid lookups that we've already done.  The paint property binder has to be found in a map by layer ID for each tile, and the render tile would have to be found in the collection by tile ID each time.

These lookups are pretty cheap individually, but get done a huge number of times, so I think they're worth eliminating.  Ideally, the tweakers execute many times for each render layer update.